### PR TITLE
[GPU] Support non constant input of split length for variadic split

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Support non constant input of split length for variadic split*

### Tickets:
 - *116385*

### CODE REVIEW

**Comment by andrew-k-park** [comment]
It looks like this PR was created based on another your PR(https://github.com/openvinotoolkit/openvino/pull/18630) branch.

File: src/plugins/intel_gpu/src/graph/crop.cpp [hunk 1]
```diff
@@ -63,13 +63,19 @@ std::vector<layout> crop_inst::calc_output_layouts(const crop_node& /*node*/, co
...
 
-        OPENVINO_ASSERT(impl_param.memory_deps.count(2) > 0, "[GPU] Can't find Crop(ngraph VariadicSplit op mode) split length values memory dependency");
-        auto split_length_mem = impl_param.memory_deps.at(2);
-        cldnn::mem_lock<uint8_t, mem_lock_type::read> split_length_mem_lock(split_length_mem, impl_param.get_stream());
-        const_data.emplace(2, make_host_tensor(split_length_mem->get_layout(), split_length_mem_lock.data()));
-
-        ov::op::v1::VariadicSplit op;
-        output_shapes = shape_infer(&op, input_shapes, ov::make_tensor_accessor(const_data));
+        if (impl_param.memory_deps.count(2) > 0) {
+            OPENVINO_ASSERT(impl_param.memory_deps.count(2) > 0, "[GPU] Can't find Crop(ngraph VariadicSplit op mode) split length values memory dependency");
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/graph/crop.cpp:67]: Is this assert still necessary?

**Comment by yeonbok** [comment]
LGTM if above minor comments are resolved. 
BTW make_host_tensor should be updated if this PR (https://github.com/openvinotoolkit/openvino/pull/18925/files) is merged prior to this PR

File: src/plugins/intel_gpu/src/graph/crop.cpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/crop.cpp]: deleted

**Review by yeonbok — APPROVED** [review]

**Pull request merged**